### PR TITLE
RDoc-1601: Added folder Client API > Rest API

### PR DESCRIPTION
**Pull request by morhilai**
-subfolder Document Commands
-documented CRUD and batch commands

### CODE REVIEW

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 1]
```diff
@@ -0,0 +1,854 @@
...
+    "@last-modified": "<date and time UTC>",
+    "Type": "PATCH",
+    "PatchStatus": "<patch status>",
+    "Debug": null
+}
+{CODE-BLOCK/}
+
+| Parameter | Description |
+| - | - | - |
+| **@id** | The ID of the document that has been created or deleted. |
```
Comment (maximburyak) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]:
```suggestion
| **@id** | The ID of the document that has been created. |
```

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 2]
```diff
@@ -0,0 +1,854 @@
...
+    "Hash": "<Hash>",
+    "ContentType": "<MIME type>",
+    "Size": <attachment size in bytes>,
+    "DocumentChangeVector": "<current change vector>"
+}
+{CODE-BLOCK/}
+
+| Parameter | Description |
+| - | - | - |
+| **Id** | The document ID prefix of the documents which were deleted. |
```
Comment (maximburyak) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]:
```suggestion
| **Id** | The document ID of the document to which the attachment belongs. |
```

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 3]
```diff
@@ -0,0 +1,854 @@
...
+    "@id": "<document ID>",
+    "Name": "<attachment name",
+    "DocumentChangeVector": "<current change vector>"
+}
+{CODE-BLOCK/}
+
+| Parameter | Description |
+| - | - | - |
+| **Type** | Same as the `Type` of the command sent - in this case `AttachmentDELETE`. |
+| **@id** | The ID of the documents that was deleted. |
```
Comment (maximburyak) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]:
```suggestion
| **@id** | The ID of the documents to which the deleted attachment belongs. |
```

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 4]
```diff
@@ -0,0 +1,854 @@
...
+{CODE-BLOCK: JavaScript}
+{
+    "Results":[
+        { },
+        ...
+    ]
+}
+{CODE-BLOCK/}
+
+* Format within the `Results` array in the response body:
```
Comment (maximburyak) [inline Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]: We need to state here that the returned array of responses is in the order of the received commands

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/get-documents-by-prefix.markdown [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+{CODE-BLOCK/}
+
+{PANEL/}
+
+{PANEL: Request Format}
+
+This is the general form of a cURL request that uses all parameters:  
+
+{CODE-BLOCK: batch}
+curl -X GET http://<server URL>/databases/<database name>/docs? \
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/get-documents-by-prefix.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 6]
```diff
@@ -0,0 +1,854 @@
...
+}
+{CODE-BLOCK/}
+
+| Parameter | Description | Required |
+| - | - | - |
+| **Id** | ID of document to create or update | Required |
+| **ChangeVector** | If updating an existing document - that document's expected change vector. If it does not match the server-side change vector a concurrency exception is thrown. | Optional |
+| **Document** | JSON Document to create, or to replace existing document | Required |
+| **Type** | Set to `PUT` | Required |
+| **ForceRevisionCreationStrategy** | Can be set to either `none` or `before` | Optional |
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/put-documents.markdown [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    "Id":"person/1-A",
+    "ChangeVector":"A:1"
+}
+{CODE-BLOCK/}
+
+#### 2) Update that same document.  
+
+{CODE-BLOCK: bash}
+curl -X PUT http://live-test.ravendb.net/databases/Example/docs?id=person/1-A \
+--header If-Match: A:1 \
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/put-documents.markdown]: Fixed

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/put-documents.markdown [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+{CODE-BLOCK/}
+
+{PANEL/}
+
+{PANEL: Request Format}
+
+This is the general form of a cURL request:  
+
+{CODE-BLOCK: batch}
+curl -X PUT http://<server URL>/databases/<database name>/docs?id=<document ID> \
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/put-documents.markdown]: fixed

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/get-documents-by-id.markdown [hunk 9]
```diff
@@ -0,0 +1,452 @@
...
+{CODE-BLOCK/}
+
+{PANEL/}
+
+{PANEL: Request Format}
+
+This is the general form of a cURL request that uses all parameters:  
+
+{CODE-BLOCK: batch}
+curl -X GET http://<server URL>/databases/<database name>/docs? \
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/get-documents-by-id.markdown]: fixed

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/delete-document.markdown [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+Server: nginx
+Date: Tue, 27 Aug 2019 11:40:12 GMT
+Connection: keep-alive
+Raven-Server-Version: 4.2.3.42
+{CODE-BLOCK/}
+
+### Request
+
+{CODE-BLOCK: bash}
+curl -X DELETE http://<server URL>/databases/<database name>/docs?id=<document ID> \
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/delete-document.markdown]: fixed

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 11]
```diff
@@ -0,0 +1,854 @@
...
+    "@collection": "<collection name>",
+    "@change-vector": "<current change vector>",
+    "@last-modified": "<date and time UTC>"
+}
+{CODE-BLOCK/}
+
+| Parameter | Description |
+| - | - | - |
+| **Type** | Same as the `Type` of the command sent - in this case `PUT`. |
+| **@id** | The ID of the document that has been created or deleted. |
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]: Fixed

**Comment by arekpalinski** [comment]
@maximburyak Can you please verify all changes you requested have been applied?

**Comment by morhilai** [comment]
@arekpalinski there are still changes to be applied.

**Comment by arekpalinski** [comment]
ok, thanks

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown (hunk 4 above)
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]: Done

File: Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown [hunk 12]
```diff
@@ -0,0 +1,854 @@
...
+| Parameter | Description | Required |
+| - | - | - |
+| **Id** | ID of document for which to delete the attachment | Required |
+| Name | Name of the attachment to delete | Required |
+| **ChangeVector** | The document's expected change vector. If it does not match the server-side change vector a concurrency exception is thrown. | Optional |
+| **Type** | Set to `AttachmentDELETE` | Required |
+
+---
+
+### Batch Options
```
[the remark this answers is not in the document]
Comment (morhilai) [reply Documentation/4.2/Raven.Documentation.Pages/client-api/rest-api/document-commands/batch-commands.markdown]: Sounds good. Done

File: Raven.Documentation.Web/Web.config [hunk 13]
```diff
@@ -1,80 +1,88 @@
...
+    For a description of web.config changes see http://go.microsoft.com/fwlink/?LinkId=235367.
+
+    The following attributes can be set on the <httpRuntime> tag.
+      <system.Web>
+        <httpRuntime targetFramework="4.6.1" />
+      </system.Web>
+  -->
   <system.web>
-    <compilation debug="true" targetFramework="4.7.1">
+    <compilation debug="true" targetFramework="4.6.1">
```
Comment (arekpalinski) [inline Raven.Documentation.Web/Web.config]: Can you please revert changes in this file. In particular the above one?

File: Raven.Documentation.Web/Raven.Documentation.Web.csproj [hunk 14]
```diff
@@ -13,7 +13,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Raven.Documentation.Web</RootNamespace>
     <AssemblyName>Raven.Documentation.Web</AssemblyName>
-    <TargetFrameworkVersion>v4.7.1</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Comment (arekpalinski) [inline Raven.Documentation.Web/Raven.Documentation.Web.csproj]: Can you please revert this?

File: Raven.Documentation.Web/Web.config [hunk 15]
```diff
@@ -1,80 +1,88 @@
...
   <system.web>
     <compilation debug="true" targetFramework="4.7.1">
       <assemblies>
-        <add assembly="netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51" />
+        <add assembly="netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51"/>
       </assemblies>
     </compilation>
-    <httpRuntime targetFramework="4.7.1" />
-    <globalization uiCulture="en-US" culture="en-US" />
+    <httpRuntime targetFramework="4.6.1"/>
```
Comment (arekpalinski) [inline Raven.Documentation.Web/Web.config]: we still have this

Comment (morhilai) [reply Raven.Documentation.Web/Web.config]: Fixed

**Pull request merged**